aiozipkin/transport.py

```diff
@@ -38,34 +45,63 @@ def send(self, record: Record) -> None:
         pass
 
 
-class Transport(TransportABC):
+class BatchManager:
 
-    def __init__(self, address: str, send_interval: float = 5,
-                 loop: OptLoop = None) -> None:
-        self._address = URL(address)
-        self._queue = []  # type: List[Dict[str, Any]]
-        self._closing = False
+    def __init__(self, max_size: int, send_interval: float,
+                 attempt_count: int, send_data: SendDataCoro,
+                 loop: asyncio.AbstractEventLoop) -> None:
+        self._max_size = max_size
         self._send_interval = send_interval
-        self._loop = loop or asyncio.get_event_loop()
-        self._session = aiohttp.ClientSession(loop=self._loop)
+        self._send_data = send_data
+        self._attempt_count = attempt_count
+        self._loop = loop
+        self._sending_batches = []  # type: List[Tuple[int, DataList]]
+        self._active_batch = None  # type: Optional[DataList]
+        self._ender = self._loop.create_future()
+        self._timer = None  # type: Optional[asyncio.Future[Any]]
         self._sender_task = asyncio.ensure_future(
             self._sender_loop(), loop=self._loop)
 
-        self._ender = self._loop.create_future()
-        self._timer = None  # type: Optional[asyncio.Future[Any]]
+    def add(self, data: Dict[str, Any]) -> None:
+        if self._active_batch is None:
+            self._active_batch = []
+        self._active_batch.append(data)
+        if len(self._active_batch) >= self._max_size:
+            self._sending_batches.append((0, self._active_batch))
```

Comment:
Should we also restrict size of self._sending_batches? This number may be very large, but still we need limit. Say zipkin server died ore restarted for some reason, and we do all this retries with timeouts, in same time `self._sending_batches` can be populated very quickly on live system. 

I think we need set limit on queue and send only recent data if possible.  This is very easy to do with `collections.deque(maxlen=10**4)`:

https://docs.python.org/3/library/collections.html#collections.deque